src/wp-includes/theme.php

```diff
@@ -840,6 +841,26 @@ function switch_theme( $stylesheet ) {
 	$new_theme->delete_pattern_cache();
 	$old_theme->delete_pattern_cache();
 
+	// Set autoload=no for the previous all themes.
+	$theme_mods_compare_value = '%' . $wpdb->esc_like( 'theme_mods_' ) . '%';
+	$theme_mods_options       = $wpdb->get_results( $wpdb->prepare( "SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name LIKE %s", $theme_mods_compare_value ) );
+
+	// Loop through fetched options to manage autoload settings.
+	foreach ( (array) $theme_mods_options as $option ) {
+		if ( "theme_mods_$stylesheet" === $option->option_name ) {
+			continue;
+		}
+
+		// Disable autoload for previous all themes.
+		wp_set_option_autoload( $option->option_name, 'no' );
+	}
+
+	// Set autoload=yes for the switched theme if not already set.
+	wp_set_option_autoload( "theme_mods_$stylesheet", 'yes' );
```

Comment:
This could be simplified and made more efficient with a single `UPDATE` query by using `wp_set_option_autoload_values()`:

```suggestion
	// Set autoload=no for the previous all themes.
	$theme_mods_options = $wpdb->get_col( 
		$wpdb->prepare( 
			"SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name != %s AND option_name LIKE %s", 
			"theme_mods_$stylesheet", 
			$wpdb->esc_like( 'theme_mods_' ) . '%' 
		) 
	);

	$autoload = array_fill_keys( $theme_mods_options, 'no' );
	$autoload[ "theme_mods_$stylesheet" ] = 'yes';
	wp_set_option_autoload_values( $autoload );
```